gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/AuthenticationFlowHandlerTest.java

```diff
@@ -227,26 +252,29 @@ public void shouldNoRedirectToMFAChallengePage_adaptiveMFA_no_active_enroll_and_
                 },
                 HttpStatusCode.FOUND_302, "Found", null);
     }
+
     @Test
     public void shouldRedirectToMFAChallengePage_adaptiveMFA_no_active_enroll_and_endUser_is_enrolling() throws Exception {
         router.route().order(-1).handler(rc -> {
+            String rule = "{context.attributes['geoip']['country_iso_code'] == 'FR'";
+            EnrollSettings enrollSettings = createEnrollSettings(true, false, MfaEnrollType.REQUIRED, 0, "");
+            ChallengeSettings challengeSettings = createChallengeSettings(true, MfaChallengeType.REQUIRED, rule);
```

Comment:
**todo** : based on the test case name, the test is related to adaptive MFA but the Challenge type is REQUIRED, should be CONDITIONAL